<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Darwin would be proud</string>
  </property>
  <widget class="QWidget" name="MainContent">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QGridLayout" name="CanvasParent"/>
    </item>
   </layout>
  </widget>
  <widget class="QDockWidget" name="Dock">
   <property name="minimumSize">
    <size>
     <width>300</width>
     <height>487</height>
    </size>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="DockContents">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="DockBody" stretch="0">
       <property name="leftMargin">
        <number>8</number>
       </property>
       <item>
        <widget class="QTabWidget" name="mainTabWidget">
         <property name="currentIndex">
          <number>1</number>
         </property>
         <widget class="QWidget" name="shaderTab">
          <attribute name="title">
           <string>Shaders</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QCheckBox" name="useLightingForShaders">
             <property name="text">
              <string>Use Lighting (point)</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="useOrbitingCamera">
             <property name="text">
              <string>Use Orbiting Camera</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="InitializeAndSrc" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="initializeShaderSceneButton">
                <property name="text">
                 <string>Initialize!</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="showShaderSourceButton">
                <property name="text">
                 <string>Show src</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="SelectionPane">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <item>
               <widget class="QLabel" name="SelectionPaneTitle">
                <property name="text">
                 <string>Selection Pane</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <layout class="QGridLayout" name="SelectionOptions">
                <property name="sizeConstraint">
                 <enum>QLayout::SetDefaultConstraint</enum>
                </property>
                <property name="spacing">
                 <number>0</number>
                </property>
                <item row="2" column="0">
                 <widget class="QCheckBox" name="shader4Box">
                  <property name="text">
                   <string>4</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QCheckBox" name="shader3Box">
                  <property name="text">
                   <string>3</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QCheckBox" name="shader1Box">
                  <property name="text">
                   <string>1</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QCheckBox" name="shader2Box">
                  <property name="text">
                   <string>2</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QCheckBox" name="shader5Box">
                  <property name="text">
                   <string>5</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="2">
                 <widget class="QCheckBox" name="shader6Box">
                  <property name="text">
                   <string>6</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <widget class="QWidget" name="widget" native="true">
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <property name="spacing">
                  <number>0</number>
                 </property>
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QPushButton" name="SelectAll">
                   <property name="text">
                    <string>Select all</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QPushButton" name="SelectNone">
                   <property name="text">
                    <string>Select none</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QToolBox" name="MainToolbox">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Ignored" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <widget class="QWidget" name="NewShaders">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>269</width>
                <height>213</height>
               </rect>
              </property>
              <attribute name="label">
               <string>Replace Shaders</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout">
               <item>
                <widget class="QPushButton" name="RefreshButton">
                 <property name="text">
                  <string>Refresh Population</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="RefreshInstructions">
                 <property name="text">
                  <string>Selected Shaders will be replaced with random new ones.</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                 <property name="wordWrap">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="replaceWithMutations">
                 <property name="text">
                  <string>Replace with Mutations</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="MutInstructions">
                 <property name="text">
                  <string>The selected shaders will be repalced with mutated versoins of a &quot;Doner&quot; Shader</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                 <property name="wordWrap">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QWidget" name="DonorSelector" native="true">
                 <layout class="QHBoxLayout" name="horizontalLayout_4">
                  <item>
                   <widget class="QLabel" name="DonorLabel">
                    <property name="text">
                     <string>Donor</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="comboBox">
                    <item>
                     <property name="text">
                      <string>1</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>2</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>3</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>4</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>5</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>6</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="toolBoxPage2">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>269</width>
                <height>213</height>
               </rect>
              </property>
              <attribute name="label">
               <string>Mutate Current Shaders</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_4">
               <item>
                <widget class="QPushButton" name="MutateSelected">
                 <property name="text">
                  <string>Mutate</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="MutationINstructions">
                 <property name="text">
                  <string>All the selected shaders will undergo one mutation.</string>
                 </property>
                 <property name="wordWrap">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
             <widget class="QWidget" name="toolBoxPage3">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>269</width>
                <height>213</height>
               </rect>
              </property>
              <attribute name="label">
               <string>Create Offspring</string>
              </attribute>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="QWidget" name="Parent1Selector" native="true">
                 <layout class="QHBoxLayout" name="horizontalLayout_6">
                  <item>
                   <widget class="QLabel" name="ParentLabel">
                    <property name="text">
                     <string>Parent 1</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="Parent1Chooser">
                    <item>
                     <property name="text">
                      <string>1</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>2</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>3</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>4</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>5</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>6</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item>
                <widget class="QWidget" name="Parent2Selector" native="true">
                 <layout class="QHBoxLayout" name="horizontalLayout_7">
                  <item>
                   <widget class="QLabel" name="ParentLabel_2">
                    <property name="text">
                     <string>Parent 2</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QComboBox" name="Parent2Chooser">
                    <item>
                     <property name="text">
                      <string>1</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>2</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>3</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>4</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>5</string>
                     </property>
                    </item>
                    <item>
                     <property name="text">
                      <string>6</string>
                     </property>
                    </item>
                   </widget>
                  </item>
                 </layout>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="Offspringbutton">
                 <property name="text">
                  <string>Create Offspring</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="OffspringInstr">
                 <property name="text">
                  <string>Selected Shaders will be replaced with offspring of Parent 1 and 2. Genes from one parent will be Gen 1; other Gen 0.</string>
                 </property>
                 <property name="wordWrap">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="importShaderTab">
          <attribute name="title">
           <string>Import</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>You can make your own shaders, or paste in the souce of previous shaders you've seen and liked.</string>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="orbitingCameraImport">
             <property name="text">
              <string>Use Orbiting Camera</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="importTextArea"/>
           </item>
           <item>
            <widget class="QWidget" name="widget_2" native="true">
             <layout class="QHBoxLayout" name="horizontalLayout_9">
              <item>
               <widget class="QLabel" name="shapeLabel">
                <property name="text">
                 <string>Shape:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="shapeSelector">
                <item>
                 <property name="text">
                  <string>Cube</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Cone</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Cylinder</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Torus</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>Sphere</string>
                 </property>
                </item>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="importButton">
             <property name="text">
              <string>Import</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="LsystemTab">
          <attribute name="title">
           <string>L-system Trees</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_7">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QComboBox" name="lSystemType">
             <item>
              <property name="text">
               <string>2D Tree</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Seaweed</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2D Maple</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bad Tree</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Bush</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>Simple 3D</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>3D Tree</string>
              </property>
             </item>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="lengthStochasticity">
             <property name="text">
              <string>Length Stochasticity</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="angleStochasticity">
             <property name="text">
              <string>Angle Stochasticity</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="leaves">
             <property name="text">
              <string>Leaves</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Recursive Depth</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSlider" name="recursiveDepth">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>5</number>
             </property>
             <property name="sliderPosition">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="tickPosition">
              <enum>QSlider::TicksBelow</enum>
             </property>
             <property name="tickInterval">
              <number>1</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="generateLSystemTree">
             <property name="text">
              <string>Generate!</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="useOrbitingCameraLTrees">
             <property name="text">
              <string>Use Orbiting Camera</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="combinedTab">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <attribute name="title">
           <string>Gallery</string>
          </attribute>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QTextBrowser" name="textBrowser">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="html">
              <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.AppleSystemUIFont'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-weight:600;&quot;&gt;The Beagle Exhibit&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-weight:600;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;White and Addo exploring the intersection of evolution and recursion.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;circa 2020&lt;/p&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Price tag: $200,000&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="regenerateGallery">
             <property name="text">
              <string>Generate!</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="useOrbitingCameraGallery">
             <property name="text">
              <string>Use Orbiting Camera</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
